ibrcommon/ibrcommon/thread/RWMutex.cpp

```diff
@@ -21,17 +21,58 @@
 
 #include "ibrcommon/thread/RWMutex.h"
 #include <errno.h>
+#include <cstring>
 
 namespace ibrcommon
 {
 	RWMutex::RWMutex()
 	{
-		pthread_rwlock_init(&_rwlock, NULL);
+		std::memset(&_rwlock,0,sizeof(pthread_rwlock_t));
+		
+		switch(pthread_rwlock_init(&_rwlock, NULL))
```

Comment:
This is bad practice.
Exceptions thrown by constructors are hard to catch. This is especially true if the objects are initialized via an initialization list.